test/mockupdb/test_query_read_pref_sharded.py

```diff
@@ -53,19 +53,12 @@ def test_query_and_read_mode_sharded_op_query(self):
                 collection = client.db.get_collection('test',
                                                       read_preference=mode)
                 cursor = collection.find(query.copy())
-                with going(next, cursor):
-                    request = server.receives()
-                    if mode in modes_without_query:
-                        # Filter is hoisted out of $query.
-                        request.assert_matches(Command(find_command))
-                        self.assertFalse('$readPreference' in request)
-                    else:
-                        # Command is nested in $query.
-                        request.assert_matches(Command(
-                            SON([('$query', find_command),
-                                 ('$readPreference', mode.document)])))
 
-                    request.replies({'cursor': {'id': 0, 'firstBatch': [{}]}})
+                command = OpMsg(find_command)
+
+                go(next, cursor)
+                request = server.receives(command)
+                request.replies({'cursor': {'id': 0, 'firstBatch': [{}]}})
 
     @unittest.skipUnless(version_tuple >= (3, 7), "requires PyMongo 3.7")
```

Comment:
Please search for `version_tuple` in test/mockupdb/ and remove all code that uses pymongo's version and remove any dead code. Now that the tests live in pymongo itself we always know the version.